### PR TITLE
feat: modelKeyRegistry and codecRegistry proper implementation

### PR DESCRIPTION
**Pull request by saskaale**

### CODE REVIEW

File: src/transport/registry/modelKeyRegistry.h [hunk 1]
```diff
@@ -2,35 +2,83 @@
...
+             * @brief Get last codecKey for specified modelKey
+             * 
+             * Search in the bi-directional map for specified modelKey
+             * 
+             * @param modelKey String representation of the model key to be searched
+             * @param codecKey output parameter where is the codecKey returned
+             * 
+             * @returns true if the modelKey has been found in map, false otherwise
+             */
+            bool tryGetCodecKey(const ModelKey& modelKey, std::string& codecKey);
```
Comment (peter-quix) [inline src/transport/registry/modelKeyRegistry.h:61]: I had a look at the original code, I think this to replace the GetType. I believe it is not actually used anywhere, so we can keep just `tryGetModelKey` and `registerModel`.

Likely something that was initially used then refactored away.

**Review by peter-quix — CHANGES_REQUESTED** [review]
Looks ok at first glance. Left 2 comments. Let me know once done so we can merge this.

File: src/transport/modelKey.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Class implements abstraction over the model key.
+     * Implemented as future preparation for the extension ( e.g. versioning )
+     * For now simply wraps the string modelKey.
+     */
+    class ModelKey{
+        private:
+            std::string key_;
+        public:
+            inline ModelKey(const std::string& modelKey = "UNKNOWN"):key_(modelKey){};
```
[the remark this answers is not in the document]
Comment (saskaale) [reply src/transport/modelKey.h:16]: in order for class being used inside std::containers it needs to be able to being initialized empty hence this default constructor with no perameters

File: src/transport/registry/modelKeyRegistry.h (hunk 1 above)
Comment (saskaale) [reply src/transport/registry/modelKeyRegistry.h:61]: okay, yes it was supposed to replace `GetType`, which did not made sense from the naming perspective.

Spaking about usage, you are right. Idouble checked it and it seemed to be refactored away.

**Review by peter-quix — APPROVED** [review]

**Pull request merged**